nvi

```diff
@@ -0,0 +1,187 @@
+    configurable using options flags.
+
+    nvi saves executable files on your disk, which lets you put these files on
+    the \$PATH and refer to them in CI environments where bash functions and
+    other popular ways of providing node runtimes are cumbersome to work with.
+
+${B}OPTIONS${N}
+    ${U}Informational${N} (just print to stdout and exit)
+    -h, --help
+        Print this usage guide.
+    -v, --version
+        Print the currently installed version of nvi.
+
+    ${U}Operational${N} (will affect the installation)
+    -d, --download-directory
+        Make nvi store the temporary tar (which is later removed) in this
+        directory. Defaults to ${B}\$PWD${N}.
+    -e, --executable-directory
+        Make nvi link the node and npm executables in this directory. Defaults
+        to ${B}\$HOME/.local/bin${N}.
+    -i, --install-directory
+        Make nvi unpack the Node.js tar in this directory. Defaults to
+        ${B}\$HOME/.local${N}. This is not removed as the executables need it.
+    -n, --node-version
+        Make nvi install this version of Node.js. Defaults to reading
+        ${B}./package.json${N} and using the value of the property ${B}.engines.node${N}.
+
+${B}EXAMPLES${N}
+    The examples range from very explicit to very implicit. Each option can be
+    omitted, which in most cases just use a conservative fallback that should be
+    safe. In the case of the Node.js version, however, the fallback is to read
+    ${B}./package.json${N} to infer the version based on the property ${B}.engines.node${N}.
+    If you are specifying your Node.js version in some other way, you will have
+    to explicitly pass the version you want using ${U}--node-version${N}.
+
+    Install Node.js v6.10.0 in \$HOME/node with executables in \$HOME/node/bin,
+    using /tmp for the temporary tar file:
+
+        $ nvi -d /tmp -i \$HOME/node -e \$HOME/node/bin -n 6.10.0
+
+    Install Node.js v6.10.0 in \$HOME/node with executables in \$HOME/node/bin:
+
+        $ nvi -i \$HOME/node -b \$HOME/node/bin -n 6.10.0
+
+    Install Node.js in the default dirs (\$HOME/.local and \$HOME/.local/bin):
+
+        $ nvi -n 6.10.0
+
+    Install Node.js using package.json in the default dirs:
+
+        $ nvi
+
+${B}SEE ALSO${N}
+    ${U}nvm${N} (https://github.com/creationix/nvm)
+        A popular tool for managing Node.js versions during development. This
+        provides a lot more functionality in the "managing" section, but exposes
+        everything as bash functions, which can cause headaches in CI
+        environments. If your main concern is managing Node.js versions in your
+        local bash sessions on your own machine, this could be worth looking
+        into.
+
+EOF
+}
+
+function install_node() {
+  NODE_VERSION=$1
+  DOWNLOAD_DIR=$2
+  INSTALL_DIR=$3
+  BIN_DIR=$4
+  [ ! -d $DOWNLOAD_DIR ] && FRESH_DOWNLOAD_DIR=1
+
+  # Ensure the directories exist.
+  mkdir --parents $DOWNLOAD_DIR
```

Comment:
Perhaps we should add a `--` after the command flags to ensure people do not accidentally send flags along to these commands. The same probably goes for the `cp` and `ln`commands further down.